pennylane/pauli/pauli_arithmetic.py

```diff
@@ -0,0 +1,279 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The Pauli arithmetic abstract reduced representation classes"""
+from copy import copy
+
+import numpy as np
+from scipy import sparse
+from pennylane import math, wires
+
+I = "I"
+X = "X"
+Y = "Y"
+Z = "Z"
+
+matI = np.eye(2)
+matX = np.array([[0, 1], [1, 0]])
+matY = np.array([[0, -1j], [1j, 0]])
+matZ = np.array([[1, 0], [0, -1]])
+
+mat_map = {
+    I: matI,
+    X: matX,
+    Y: matY,
+    Z: matZ,
+}
+
+sparse_matI = sparse.eye(2, format="csr")
+sparse_matX = sparse.csr_matrix([[0, 1], [1, 0]])
+sparse_matY = sparse.csr_matrix([[0, -1j], [1j, 0]])
+sparse_matZ = sparse.csr_matrix([[1, 0], [0, -1]])
+
+sparse_mat_map = {
+    I: sparse_matI,
+    X: sparse_matX,
+    Y: sparse_matY,
+    Z: sparse_matZ,
+}
+
+_map_I = {
+    I: (1, I),
+    X: (1, X),
+    Y: (1, Y),
+    Z: (1, Z),
+}
+_map_X = {
+    I: (1, X),
+    X: (1, I),
+    Y: (1.0j, Z),
+    Z: (-1.0j, Y),
+}
+_map_Y = {
+    I: (1, Y),
+    X: (-1.0j, Z),
+    Y: (1, I),
+    Z: (1j, X),
+}
+_map_Z = {
+    I: (1, Z),
+    X: (1j, Y),
+    Y: (-1.0j, X),
+    Z: (1, I),
+}
+
+mul_map = {I: _map_I, X: _map_X, Y: _map_Y, Z: _map_Z}
+
+
+class PauliWord(dict):
+    """Immutable dictionary used to represent a Pauli Word.
+    Can be constructed from a standard dictionary.
+
+    >>> w = PauliWord({"a": X, 2: Y, 3: Z})
+    """
+
+    def __missing__(self, key):
+        """If the wire is not in the Pauli word,
+        then no operator acts on it, so return the Identity."""
+        return I
+
+    def __init__(self, mapping={}):  # pylint:disable=dangerous-default-value
```

Comment:
```suggestion
    def __init__(self, mapping):
```

How about just leave it as a mandatory argument?

Reply:
Makes sense!